distribution/kernel-debuginfo/runtest.sh

```diff
@@ -0,0 +1,145 @@
+#!/bin/bash
+set -x
+
+. /usr/bin/rhts_environment.sh
+
+function test_pass()
+{
+	report_result $TEST PASS 0
+}
+
+function test_fail()
+{
+	report_result $TEST FAIL 1
+}
+
+function nvr_kernel_current()
+{
+	rpm -qa "kernel*" | grep "$(uname -r)" | sort | head -n1
+}
+
+function nvr_kernel_latest()
+{
+	yum list --available kernel{,-rt}.$(uname -m) --showduplicates \
+	| sed "s/.$(uname -m)//" \
+	| sort -k1,1 -k2V \
+	| awk 'END { print $1"-"$2".'$(uname -m)'"; }'
+}
+
+# Determine which kernel debuginfo to download.
+# By default, use currently running kernel (unless overriden from env).
+# This can be achieved either by specifying KERNEL_NVR directly, or one
+# of the following two test config options:
+# CONFIG_NVR_KERNEL_{CURRENT,LATEST}
+
+if [ -z "${KERNEL_NVR++}" ]
+then
+	echo "DEBUG: Using current kernel nvr."
+	KERNEL_NVR="$(nvr_kernel_current)"
+else
+	echo "DEBUG: Using user provided kernel NVR: $KERNEL_NVR"
+fi
+
+if [ -n "$CONFIG_NVR_KERNEL_CURRENT" ]
+then
+	echo "DEBUG: Using current kernel nvr."
+	KERNEL_NVR="$(nvr_kernel_current)"
+fi
+
+if [ -n "$CONFIG_NVR_KERNEL_LATEST" ]
+then
+	echo "DEBUG: Using latest available kernel."
+	KERNEL_NVR="$(nvr_kernel_latest)"
+fi
+
+if [ -z "${KERNEL_NVR++}" ]
+then
+	echo "ERROR: Unable to determine kernel nvr."
+	echo "DEBUG: CONFIG_NVR_KERNEL_CURRENT: " \
+	     $CONFIG_NVR_KERNEL_CURRENT
+	echo "DEBUG: CONFIG_NVR_KERNEL_LATEST: " \
+	     $CONFIG_NVR_KERNEL_LATEST
+	echo "DEBUG: rpm -qa \"kernel*\""
+	rpm -qa "kernel*"
+	echo "DEBUG: uname -r: $(uname -r)"
+	echo "DEBUG: yum list --available kernel{,rt}.$(uname -m)" \
+	     "--showduplicates:"
+	yum list --available kernel{,-rt}.$(uname -m) --showduplicates
+	test_fail "FAILED. SEE ERROR OUTPUT ABOVE."
```

Comment:
Should this rather abort or skip? Similar with some other cases (e.g. with inability to download debuginfo RPM), the lack of the package would signify a problem in the pipeline rather then a kernel bug that should be reported to patch authors.

Reply:
Sure. Looking at valid results for rhts-report-result, "Valid values are PASS, WARN, FAIL, or SKIP."; so I can just use
```
function test_skip() { report_result $TEST SKIP 0; }
# ...
test_skip ...
```
Or would you prefer WARN?

Reply:
We use WARN/ABORTED to mark infrastructure issues -- e.g. the test can't be compiled, the network is down in the lab.

We use SKIP/COMPLETED to mark the test requirements are not met (e.g. a debug package is required for the test but we don't build it).

So in this case, SKIP seems to be making more sense.

Reply:
Changes present in v2.